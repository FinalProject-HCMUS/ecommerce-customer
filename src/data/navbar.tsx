export const navbarLinks = [
  {
    label: 'Products',
    path: '/search',
    authenticate: false, // No authentication required
  },
  {
    label: 'Chat',
    path: '/chat',
    authenticate: true, // No authentication required
  },
  {
    label: 'Policy',
    path: '/policy',
    authenticate: false, // No authentication required
  },
  {
    label: 'Blog',
    path: '/blog',
    authenticate: false, // No authentication required
  },
  {
    label: 'Orders',
    path: '/orders',
    authenticate: true, // Requires authentication
  },
]

export const shopName = 'ShopName'

export const navbarSearchPlaceholder = 'Search for products...'
