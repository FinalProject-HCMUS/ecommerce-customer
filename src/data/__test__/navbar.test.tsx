import { navbarLinks, shopName, navbarSearchPlaceholder } from '../navbar';

describe('navbar data', () => {
  test('navbarLinks should be an array with correct structure', () => {
    expect(Array.isArray(navbarLinks)).toBe(true);
    expect(navbarLinks.length).toBeGreaterThan(0);

    navbarLinks.forEach(link => {
      expect(link).toHaveProperty('label');
      expect(typeof link.label).toBe('string');
      expect(link).toHaveProperty('path');
      expect(typeof link.path).toBe('string');
      expect(link).toHaveProperty('authenticate');
      expect(typeof link.authenticate).toBe('boolean');
    });

    // Check for some expected links
    expect(navbarLinks.map(l => l.label)).toEqual(
      expect.arrayContaining(['products', 'chat', 'policy', 'blog', 'orders'])
    );
  });

  test('shopName should be a string', () => {
    expect(typeof shopName).toBe('string');
    expect(shopName).toBe('ShopName');
  });

  test('navbarSearchPlaceholder should be a string', () => {
    expect(typeof navbarSearchPlaceholder).toBe('string');
    expect(navbarSearchPlaceholder).toBe('Search for products...');
  });

  test('should match snapshot', () => {
    expect({ navbarLinks, shopName, navbarSearchPlaceholder }).toMatchSnapshot();
  });
});