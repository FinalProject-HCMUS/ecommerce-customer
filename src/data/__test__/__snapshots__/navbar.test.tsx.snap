// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar data should match snapshot 1`] = `
{
  "navbarLinks": [
    {
      "authenticate": false,
      "label": "products",
      "path": "/products",
    },
    {
      "authenticate": true,
      "label": "chat",
      "path": "/chat",
    },
    {
      "authenticate": false,
      "label": "policy",
      "path": "/policy",
    },
    {
      "authenticate": false,
      "label": "blog",
      "path": "/blog",
    },
    {
      "authenticate": true,
      "label": "orders",
      "path": "/orders",
    },
  ],
  "navbarSearchPlaceholder": "Search for products...",
  "shopName": "ShopName",
}
`;
